source/llassetgen-rendering/main.cpp

```diff
@@ -0,0 +1,383 @@
+#include <algorithm>
+#include <iostream>
+#include <string>
+
+#include <ft2build.h>  // NOLINT include order required by freetype
+#include FT_FREETYPE_H
+
+#include <QApplication>
+#include <QBoxLayout>
+#include <QFormLayout>
+#include <QGroupBox>
+#include <QLabel>
+#include <QLineEdit>
+#include <QMainWindow>
+#include <QResizeEvent>
+#include <QValidator>
+
+#include <glm/glm.hpp>
+
+#include <glbinding/ContextInfo.h>
+#include <glbinding/Version.h>
+#include <glbinding/gl/gl.h>
+
+#include <globjects/base/File.h>
+#include <globjects/globjects.h>
+#include <globjects/logging.h>
+
+#include <globjects/Buffer.h>
+#include <globjects/Program.h>
+#include <globjects/Shader.h>
+#include <globjects/Texture.h>
+#include <globjects/VertexArray.h>
+#include <globjects/VertexAttributeBinding.h>
+
+#include <llassetgen/llassetgen.h>
+
+#include "WindowQt.h"
+
+using namespace gl;
+
+/* Taken from cginternals/globjects,
+ * example: qtexample, texture
+ * edited
+ */
+
+/*
+ * This class is used to display GL rendered content in a Qt Window.
+ * It overrides all functions from WindowQt which use GL functionality.
+ * It handles the GL context and loads the shader from file.
+ * Some functions are connected to the GUI using Qt signals and slots.
+ */
+class Window : public WindowQt {
+   public:
+    explicit Window(const QSurfaceFormat &format) : WindowQt(format) {
+        m_samplerIndex = 0;
+        m_texture = nullptr;
+        m_backgroundColor = glm::vec4(1.f, 1.f, 1.f, 1.f);
+        m_fontColor = glm::vec4(0.f, 0.f, 0.f, 1.f);
```

Comment:
As far as i know it is preferred to initialize these directly where the member variable is declared (i.e. `glm::vec4 m_fontColor {0.f, 0.f, 0.f, 1.f}` at the end of the class)

Reply:
Ok :)